Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -382,9 +329,9 @@ public actor SwiftLanguageServer: ToolchainLanguageServer {
 
```

Comment:
Do you have a TODO anywhere for changing this to the diagnostic request + a separate semantic token request (possibly part of enabling cancellation)?

Reply:
There already exists the separate diagnostic request https://github.com/apple/sourcekit-lsp/blob/1a7939304b3caf4680945272ac70e2dae94fdf24/Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift#L1320-L1351.

We won’t be able to get rid of the diagnostic pushing though, because most editor (AFAIK every editor except VS Code) doesn’t support the pull diagnostics request. I added an item to my To Do list to define the diagnostic push in terms of the diagnostic pull internally though. That way we can hopefully make the document update syntactic only. I will file GitHub issues for all the items on my to do list once I’m done reaving havoc around the entire code base and it’s actually possible for multiple people to work on sourcekit-lsp again.

Reply:
> I added an item to my To Do list to define the diagnostic push in terms of the diagnostic pull internally though

👍